superset/assets/src/query/buildQueries.ts

```diff
@@ -0,0 +1,18 @@
+import { FormData, getGranularity } from './formData';
+
+// TODO: fill out the rest of the query object
+export interface QueryObject {
+  granularity: string;
+  groupby?: string[];
+}
+
+// Build the common segments of the query objects (e.g. the granularity field derived from
+// either sql alchemy or druid). The segments specific to each viz type is constructed in the
+// buildQuery method for each viz type (see `wordcloud/buildQuery.ts` for an example).
+// Note the type of the formData argument passed in here is the type of the formData for a
+// specific viz, which is a subtype of the generic formData shared among all viz types.
+export default function buildQueries<T extends FormData>(formData: T): QueryObject[] {
```

Comment:
When would this return multiple query objects?

Reply:
^I think that's correct. E.g., time series with time shift is 2 queries on the backend now, so I think it will become 2 queries on the client.

Reply:
Good question. This was also bothering me as I was writing this method up. It seems each viz-specific `buildQuery` would need to provide the query count to the generic one the # of common queries to construct.

Thinking restructuring this..

Reply:
ah good question. I guess for a time shift it would be plausible to auto-generate the second query, but still potentially tricky 🤔